livekit_models.proto

```diff
@@ -167,3 +167,48 @@ message ClientInfo {
   string browser = 7;
   string browser_version = 8;
 }
+
+// server provided client configuration
+message RTCClientConfiguration {
+	AudioConfiguration audio = 1;
+  VideoConfiguration video = 2;
+  SignalConfiguration signal = 4;
+}
+
+message AudioConfiguration {
+  message AudioProcessing {
+    // echo cancellation
+    bool aec = 1;
+    // noise suppression
+    bool ns = 2;
+    // automatic gain control
+    bool agc = 3;
+  }
+  message AudioPreferences {
+    bool dtx = 1;
+    int32 sample_rate = 2;
+    int32 channel_count = 3;
+  }
+  Codec codec = 1;
+  AudioProcessing processing = 2;
+  AudioPreferences preferences = 3;
+
+  // TODO sample/playout config(micphone, speaker, earphone ...)
+}
+
+message VideoConfiguration {
+  message VideoProcessing {
+    bool hardware_decoder = 1;
+    bool hardware_encoder = 2;
+  }
+  message VideoPreferences {
+    string facing_mode = 1;
+  }
+  Codec preferred_codec = 1;
+  VideoProcessing processing = 2;
+  VideoPreferences preferences = 3;
+}
+
+message SignalConfiguration {
+	bool migration = 1;
```

Comment:
I just blindly copied this over from the proposal, what would setting `migration` do on a client? 
Enable the ability to migrate when needed ?

Reply:
I am guessing that is it @lukasIO . But, should confirmation from @cnderrauber

Reply:
Thanks, you are totally right @boks1971